### PR TITLE
Test generation w/ `--yes` flag &  TypeScript

### LINKED ISSUE #2436: 🐛 BUG: `wrangler init workername -y` adds "jest" to tsconfig.json, but nothing else
### What version of `Wrangler` are you using?

2.6.2

### What operating system are you using?

Mac

### Describe the Bug

Like the title says. I'd expect it to install `jest` to my `devDependencies`, and initialise a first test somewhere. Instead, I get this error in my tsconfig. 

image

### PR DESCRIPTION
**Pull request by JacobMGEvans**
What this PR solves:
Added support for `wrangler init` test generation and TypeScript.
The `--yes` flag now generates a test for the project using Jest and Jest types during initialization.
Test generation is also now prompted for during the generation of a TypeScript project.
    
How to test:

https://user-images.githubusercontent.com/27247160/209878819-80a1bf18-58d4-4028-bf63-32ea40cccbab.mov

Associated docs issues/PR:

- [insert associated docs issue(s)/PR(s)]

Author has included the following, where applicable:

- [x] Tests
- [x] Changeset

Reviewer has performed the following, where applicable:

- [x] Checked for inclusion of relevant tests
- [x] Checked for inclusion of a relevant changeset
- [x] Checked for creation of associated docs updates
- [x] Manually pulled down the changes and spot-tested

resolves #2436

### CODE REVIEW

File: packages/wrangler/src/init.ts [hunk 1]
```diff
@@ -515,6 +516,33 @@ export async function initHandler(args: InitArgs) {
...
+						devDepsToInstall.push(newWorkerTestType);
+						newWorkerTestType === "jest" &&
+							devDepsToInstall.push(`@types/jest`);
+
+						await writeFile(
+							path.join(creationDirectory, "./src/index.test.ts"),
+							readFileSync(
+								path.join(
+									getBasePath(),
+									`templates/init-tests/test-${newWorkerTestType}-new-worker.js`
```
Comment (rozenmd) [inline packages/wrangler/src/init.ts:534]: You're going to want to make ts-specific test templates - check out the TS example here https://developers.cloudflare.com/workers/wrangler/api/#single-worker-example

File: packages/wrangler/src/init.ts [hunk 2]
```diff
@@ -588,19 +617,12 @@ export async function initHandler(args: InitArgs) {
 						readFileSync(path.join(getBasePath(), `templates/${template}`))
 					);
 
-					logger.log(
```
Comment (rozenmd) [inline packages/wrangler/src/init.ts:591]: doesn't removing this line mean you don't get feedback after the `src/index.js` gets created?

**Comment by rozenmd** [comment]
Awesome work btw 🎉 , was going to try doing this today/tomorrow lol

File: packages/wrangler/src/init.ts (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/init.ts:591]: I moved it down, I can move it back.

**Comment by rozenmd** [comment]
You're probably going to want to split this PR in two PRs - one that fixes #2436, and another that fixes #https://github.com/cloudflare/wrangler2/issues/1774

**Comment by JacobMGEvans** [comment]
> You're probably going to want to split this PR in two PRs - one that fixes #2436, and another that fixes #1774 - there is a _crap ton_ of stuff you need to do to make jest work with TS (detailed in the issue in #1774)

Per our discussion I will be focusing on Vitest test generation in TS (since it just works) and we can kick the can for Jest test generation, until we figure out how we want to support it with ESBuild or the like. 

Cc: @rozenmd @admah

**Comment by WalshyDev** [comment]
> Per our discussion I will be focusing on Vitest test generation in TS

❤️❤️❤️❤️❤️❤️

Me complaining this would use Jest not Vitest

I am claiming credit for this 😆 

(legit though, tyyyy <3)

File: packages/wrangler/src/init.ts [hunk 3]
```diff
@@ -522,10 +523,44 @@ export async function initHandler(args: InitArgs) {
...
+					shouldCreateTests =
+						yesFlag ||
+						(await confirm(
+							"Would you like us to write your first test with Vitest?"
+						));
+
+					if (shouldCreateTests) {
+						if (yesFlag) {
+							logger.info(
+								"Currently the default TypeScript test generation is Vitest."
```
Comment (rozenmd) [inline packages/wrangler/src/init.ts:535]:
```suggestion
								"Your project will use Vitest to run your tests."
```

File: .changeset/few-pumas-retire.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+"wrangler": minor
+---
+
+Added support for `wrangler init` test generation and TypeScript.
```
Comment (rozenmd) [inline .changeset/few-pumas-retire.md:5]:
```suggestion
```

File: .changeset/few-pumas-retire.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+The `--yes` flag now generates a test for the project using Jest and Jest types during initialization.
```
Comment (rozenmd) [inline .changeset/few-pumas-retire.md:6]:
```suggestion
After this PR, `wrangler init --yes` will generate a test for your new Worker project, using Jest.
```

File: .changeset/few-pumas-retire.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+When creating a TypeScript project, the user will be prompted to generate tests with Vitest, which supports TypeScript natively.
```
Comment (rozenmd) [inline .changeset/few-pumas-retire.md:7]:
```suggestion
When using `wrangler init`, and choosing to create a Typescript project, you will now be asked if Wrangler should write tests for you, using Vitest.
```

File: .changeset/few-pumas-retire.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+This feature allows users to easily set up a testing environment for their project during the initialization process, when using the `--yes`
```
Comment (rozenmd) [inline .changeset/few-pumas-retire.md:8]:
```suggestion
```
Avoid saying anything is simple/easy/fast/quick etc - but I'd remove this sentence entirely as it repeats what we told them in previous sentences

File: .changeset/few-pumas-retire.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+flag it will default to Vitest for the TypeScript project.
+
+This resolves issue #2436.
```
Comment (rozenmd) [inline .changeset/few-pumas-retire.md:11]:
```suggestion
Closes #2436.
```

**Comment by rozenmd** [comment]
I'd change the wording in the changeset + one of the prompts, otherwise looks good to me

File: .changeset/few-pumas-retire.md (hunk 8 above)
Comment (JacobMGEvans) [reply .changeset/few-pumas-retire.md:11]: the changeset doesnt affect it being closed.

File: .changeset/few-pumas-retire.md (hunk 4 above)
Comment (JacobMGEvans) [reply .changeset/few-pumas-retire.md:5]: I prefer having a summary title.

File: .changeset/few-pumas-retire.md (hunk 7 above)
Comment (JacobMGEvans) [reply .changeset/few-pumas-retire.md:8]: Yeah, definitely should.

File: .changeset/few-pumas-retire.md (hunk 4 above)
Comment (rozenmd) [reply .changeset/few-pumas-retire.md:5]: In that case
```suggestion
Added support for `wrangler init` test generation when using TypeScript.
```

**Review by rozenmd — APPROVED** [review]
LGTM 👍

**Pull request merged**